d2bs/kolbot/libs/common/Prototypes.js

```diff
@@ -1104,3 +1104,149 @@ if (!Array.prototype.find) {
 		writable: true
 	});
 }
+
+/**
+ * @description Return the first element or undefined
+ * @return undefined|*
+ */
+if (!Array.prototype.first) {
+	Array.prototype.first = function () {
+		return this.length > 0 ? this[0] : undefined;
+	};
+}
+
+/**
+ * @description Return the items of a player, or an empty array
+ * @param args
+ * @returns Unit[]
+ */
+Unit.prototype.getItems = function (...args) {
+	let item = this.getItem.apply(this, args), items = [];
+
+	if (item) {
+		do {
+			items.push(copyUnit(item));
+		} while (item.getNext());
+	}
+
+	return items;
+};
+
+/**
+ * @description Used upon item units like ArachnidMesh.castChargedSkill([skillId]) or directly on the "me" unit me.castChargedSkill(278);
+ * @param {int} skillId = undefined
+ * @param {int} x = undefined
+ * @param {int} y = undefined
+ * @return boolean
+ * @throws Error
+ */
+Unit.prototype.castChargedSkill = function (...args) {
+	let skillId, x, y, unit, chargedItem, charge,
+		chargedItems = [],
+		validCharge = function (itemCharge) {
+			return itemCharge.skill === skillId && itemCharge.charges;
+		};
+
+	switch (args.length) {
+	case 0: // item.castChargedSkill()
+		break;
+	case 1:
+		if (args[0] instanceof Unit) { // hellfire.castChargedSkill(monster);
+			unit = args[0];
+		} else {
+			skillId = args[0];
+		}
+
+		break;
+	case 2:
+		if (typeof args[0] === 'number') {
+			if (args[1] instanceof Unit) { // me.castChargedSkill(skillId,unit)
+				[skillId, unit] = [...args];
+			} else if (typeof args[1] === 'number') { // item.castChargedSkill(x,y)
+				[x, y] = [...args];
+			}
+		} else {
+			throw new Error(' invalid arguments, expected (skillId, unit) or (x, y)');
+		}
+
+		break;
+	case 3:
+		// If all arguments are numbers
+		if (typeof args[0] === 'number' && typeof args[1] === 'number' && typeof args[2] === 'number') {
+			[skillId, x, y] = [...args];
+		}
+
+		break;
+	default:
+		throw new Error("invalid arguments, expected 'me' object or 'item' unit");
+	}
+
+	// Charged skills can only be casted on x, y coordinates
+	unit && ([x, y] = [unit.x, unit.y]);
```

Comment:
Maybe this should be an if statement, since short-circuit evaluation with assignment is a little obscure. (I'm guilty of this too, but I don't feel like it fits kolbot's style).

Reply:
If i would write it within a if, the eslinter would generate so much more code, i personally think with the comment its pretty clear what is going on,

Reply:
It's only obvious to someone more familar with Javascript at our level. It might be confusing for someone with less experience.